src/main/java/org/spongepowered/common/mixin/core/item/MixinEnchantment.java

```diff
@@ -56,17 +58,36 @@
     @Shadow public abstract String shadow$getName();
 
     private String id = "";
+    private Translation translation = null;
 
     @Inject(method = "<init>", at = @At("RETURN"))
     public void onConstructed(int id, ResourceLocation resLoc, int weight, EnumEnchantmentType type, CallbackInfo ci) {
         this.id = resLoc.toString();
+        updateTranslation();
     }
 
-    @Override
-    public String getId() {
+    @Inject(method = "setName", at = @At("RETURN"))
+    public void setName(String name, CallbackInfoReturnable<Enchantment> cir) {
+        updateTranslation();
+    }
+
+    public void updateTranslation() {
+        this.translation = new SpongeTranslation(shadow$getName());
+    }
+
+    public String enchantment$getId() {
```

Comment:
IMO its more consistent if everything is prefixed. (Both in this class and in the other classes)